pixano/datasets/builders/folders/base.py

```diff
@@ -140,28 +149,30 @@ def generate_data(
                     metadata = self._read_metadata(split / self.METADATA_FILENAME)
                 except FileNotFoundError:
                     metadata = None
-                    entities_data = None
-
-                for view_file in split.glob("*"):
-                    # only consider {split}/{item}.{ext} files
-                    if view_file.is_file() and view_file.suffix in self.EXTENSIONS:
-                        # retrieve item metadata in metadata file
-                        item_metadata = {}
-                        if metadata is not None:
-                            for m in metadata:
-                                if m[self.view_name] == view_file.name:
-                                    item_metadata = m
-                                    break
-                            if not item_metadata:
-                                raise ValueError(f"Metadata not found for {view_file}")
-
-                            # extract entity metadata from item metadata
-                            entities_data = item_metadata.pop(self.entity_name, None)
+
+                if metadata is None:
+                    for view_file in split.glob("*"):
+                        # only consider {split}/{item}.{ext} files
+                        if view_file.is_file() and view_file.suffix in self.EXTENSIONS:
```

Comment:
maybe it would be better to have a condition that is `if not view_file.is_file() or view_file.suffix not in self.EXTENSIONS: continue` and remove indentation for the yield for better readability

the legacy code was already not pretty :)